RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -366,6 +366,16 @@ class PumpOpsSynchronous {
                             events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), atIndex: 0)
                         }
                     }
+                    
+                    if let alarm = event as? PumpAlarmPumpEvent {
+                        switch alarm.alarmType {
+                        case .BatteryDepleted, .BatteryOutLimitExceeded, .DeviceReset:
+                            print("Found clock loss in pump history.  Ending history fetch.")
```

Comment:
I'm surprised this can happen while history stays intact. Alternatively the presence of this event could cause a reset of the offset interval to zero. I have a 522 that loses its clock on every battery out no matter how short.